README.md

```diff
@@ -1,52 +1,113 @@
+It uses the [yast2-registration](https://github.com/yast/yast-registration) and
+[yast2-migration](https://github.com/yast/yast-migration) modules as the backend.
+That means it provides the very same functionality which can be done using these
+two modules but in a simplified way.
 
+For migrating to SLES you need a SLES registration key, see https://www.suse.com/support/
+how to obtain it. You can create a 60-day trial activation code at
+https://www.suse.com/download/sles/ (you need to have a SUSE account for that).
 
-Development
-===========
+### Features
+
+- Simplified workflow for migrating openSUSE Leap installations
+  to SUSE Linux Enterprise (SLE) product
+- Can register against [SUSE Customer Service](https://scc.suse.com/) (SCC)
+  server or any local [RMT](https://github.com/SUSE/rmt)/SMT server
+- Supports [rollback](#rollback) when the migration fails or is aborted
+- Because openSUSE Leap shares many packages with SLE the migration should
+  usually install just few packages, it should not completely reinstall the system
+
+### Workflow
+
+1. Online update check - at the beginning the module checks whether there is
+   an update for software stack and asks to install it. This should ensure the
+   package installation later works fine. If an update for this module is needed
+   it should be marked as a software stack update as well.
+2. Entering the registration data - the module asks for the e-mail address and
+   the SLE registration code, alternatively you can enter the address of the local
+   RMT/SMT server (email and registration code are not required in that case)
+3. Registration - the system is registered to the registration server and
+   the openSUSE Leap repositories from the server are added into the system
+4. Migration - the registration is then migrated to the SLE product and the
+   Leap repositories are changed to the SLE repositories
+5. License agreement - displays the SLE product license and requires accepting it,
+   if the license is not accepted the migration is aborted (see the [rollback](
+   #rollback) section below)
+6. Migration summary - the dialog displays the summary about the installed or
+   upgraded products and packages, if there is a package dependency problem
+   then it needs to be manually resolved before continuing
+7. Package installation - the SLE packages are installed into the system, this can
+   take some time
+8. Congratulation - a congratulation dialog is displayed to confirm a successful
+   migration
+9. Reboot - it is recommended to manually reboot the system to ensure all changes
+   take effect
+
+#### Rollback
+
+When the migration fails or is aborted then YaST restores the original
+repositories in the system and moves the registration from SLE back to openSUSE
+Leap. However, the system still remains in a registered state.
+
+If you want to fully unregister the system then run this command as `root`:
+
+```shell
+SUSEConnect --de-register

```

Comment:
Question: why isn't it done as part of the rollback process?

Reply:
Unfortunately the [YaST SUSEConnect API](https://github.com/SUSE/connect-ng/blob/main/yast/lib/suse/connect/yast.rb) does not export functionality for that... :worried: 
Later I'll request that.

Reply:
Ah, ok :sob:  Thanks!